bofire/models/model.py

```diff
@@ -89,18 +89,43 @@ def _fit(self, X: pd.DataFrame, Y: pd.DataFrame):
         pass
 
     def cross_validate(
-        self, experiments: pd.DataFrame, folds: int = -1
-    ) -> Tuple[CvResults, CvResults]:
+        self,
+        experiments: pd.DataFrame,
+        folds: int = -1,
+        include_X: bool = False,
+        hooks: Dict[
```

Comment:
1. This is a complicated type definition. Maybe create custom data-class `Hook`?
2. Can the hook-name be an arbitrary string?

Reply:
1. I was not exactly sure what you meat here, we can change it later ;)
2. Yes, string is arbitrary and is the name for the hook.